installer/pyinstaller/hidden_imports.py

```diff
@@ -1,8 +1,31 @@
-from samcli.cli.command import _SAM_CLI_COMMAND_PACKAGES
+import importlib
+import pkgutil
+from typing import cast
+from typing_extensions import Protocol
 
-SAM_CLI_HIDDEN_IMPORTS = _SAM_CLI_COMMAND_PACKAGES + [
-    # terraform hook
-    "samcli.hook_packages.terraform",
+
+class HasPathAndName(Protocol):
+    __path__: str
+    __name__: str
+
+
+def walk_modules(module: HasPathAndName, visited: set) -> None:
```

Comment:
Just wondering, do we need to change anything in the hooks, or this should play well with what we already have there? https://github.com/aws/aws-sam-cli/blob/develop/installer/pyinstaller/hook-samcli.py#L10

Reply:
Good question, this is how I understand it: for py files that are used as data files (never got statically or dynamic imported), it doesn't add any value by walking through them and adding to hidden packages (we tested this before), only when we add them as data files they will work as expected. So for any module-only additions they will go into the hidden imports. But for any terraform like projects that have also py files as data files, we still need to manually add them in hook-samcli.py as data files. Any miss in that would cause pyInstaller products to miss files. So data import and the walk through logic doesn't conflict, but we do still have this manual need to update data files with code packages automatically included